chia/full_node/full_node.py

```diff
@@ -778,7 +778,11 @@ async def new_peak(self, request: full_node_protocol.NewPeak, peer: WSChiaConnec
                         full_node_protocol.RequestBlock(target_peak.height, False),
                         timeout=10,
                     )
-                    if target_peak_response is not None and isinstance(target_peak_response, RespondBlock):
+                    if (
```

Comment:
this code checks if a peer that sent us a peak that is different from our target peak has the peak that we are currently syncing to, just so we can add it to the list of peers we can use in the current sync that is running

this change looks ok, i have one question, the ABA blockchain is shorter but heavier, in that case how do they even respond to this ?

Reply:
my understanding is that they compute weight differently, in a way that inflates it, so they have higher numbers than us

Reply:
i understand now that the peers that are sending the response with the incorrect header hash are actually our peers  trying to respond for requests for the bad peak introduced from the ADA peer, feels like we should evict peaks that resulted in a bad WP, that way we wont even have these peaks in the store trying to sync to them after the first WP

Reply:
we already disconnect and ban the peer when the wp fails. new heavier peaks will keep on coming from aba nodes, so i dont think evicting peaks will help much.